libraries/chain/asset_evaluator.cpp

```diff
@@ -360,42 +360,256 @@ void_result asset_update_issuer_evaluator::do_apply(const asset_update_issuer_op
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
-void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bitasset_operation& o)
+/****************
+ * Loop through assets, looking for ones that are backed by the asset being changed. When found,
+ * perform checks to verify validity
+ *
+ * @param d the database
+ * @param op the bitasset update operation being performed
+ * @param new_backing_asset
+ * @param true if after hf 922/931 (if nothing triggers, this and the logic that depends on it
+ *    should be removed).
+ */
+void check_children_of_bitasset(database& d, const asset_update_bitasset_operation& op,
+      const asset_object& new_backing_asset, bool after_hf_922_931)
+{
+   // no need to do these checks if the new backing asset is CORE
+   if ( new_backing_asset.get_id() == asset_id_type() )
+      return;
+
+   // loop through all assets that have this asset as a backing asset
+   const auto& idx = d.get_index_type<asset_index>().indices().get<by_type>();
+
+   for( auto itr = idx.lower_bound(true); itr != idx.end(); ++itr )
+   {
+      const auto& child = *itr;
+      if ( child.bitasset_data(d).options.short_backing_asset == op.asset_to_update )
```

Comment:
For better performance, we can add a "by_backing_asset" index for `asset_bitasset_data_object` so can iterate through it here, rather then iterating through all bitAssets.

If we don't have time to do it right now, we can do it in next milestone.